### PR TITLE
chore: Harmonize Tracked Entity Controller[TECH-1571]

### PR DESCRIPTION
**Pull request by enricocolasante**
This PR creates a `TrackedEntityOperationParams` object that is used in the service layer and all the needed mappers and tests.
Same work was done [here](https://github.com/dhis2/dhis2-core/pull/14180) and [here](https://github.com/dhis2/dhis2-core/pull/14048) for events and enrollments

**Next Steps**

- Use a `TrackedEntities` object to return result from the server to hide some of the pagination complexity? (It is done this way in other entities)
- Remove `includeDeleted` from `TrackedEntityParams`
- Correclty set default values in mapper and not in `RequestParams` as Spring binding doesn't work that way

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        throws BadRequestException,
+        ForbiddenException
+    {
+        Program program = validateProgram( operationParams.getProgramUid() );
+        ProgramStage programStage = validateProgramStage( operationParams, program );
+        TrackedEntityType trackedEntityType = validateTrackedEntityType( operationParams.getTrackedEntityTypeUid() );
+
+        User user = operationParams.getUser();
+        Set<OrganisationUnit> orgUnits = validateOrgUnits( user, operationParams.getOrganisationUnits(), program );
+        if ( operationParams.getOrganisationUnitMode() == OrganisationUnitSelectionMode.CAPTURE && user != null )
```
Comment (muilpp) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java:105]: could this not be done in the method `validateOrgUnits` or in a new one?
also, ACCESSIBLE is very similar to CAPTURE mode, what happens in that case?

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/trackedentity/RequestParams.java [hunk 2]
```diff
@@ -98,7 +98,7 @@ class RequestParams extends PagingAndSortingCriteriaAdapter
      * Selection mode for the specified organisation units, default is
      * ACCESSIBLE.
      */
-    private OrganisationUnitSelectionMode ouMode;
+    private OrganisationUnitSelectionMode ouMode = OrganisationUnitSelectionMode.DESCENDANTS;
```
Comment (muilpp) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/trackedentity/RequestParams.java:101]: Why do we assume the default is DESCENDANTS?

Comment (enricocolasante) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/trackedentity/RequestParams.java]: That is a good question but I am leaving the answer to the ACL epic.
I just left everything as it was before in this PR

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java (hunk 1 above)
Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java]: Moved inside the `validateOrgUnits` method.
I guess this is not working correctly 100%, anyway, I will leave any functional change to this part for the ACL epic.

**Review by muilpp — APPROVED** [review]

**Review by zubaira — APPROVED** [review]

**Pull request merged**